rotkehlchen/api/v1/schemas.py

```diff
@@ -734,49 +739,155 @@ def generate_fields_post_validation(self, data: dict[str, Any]) -> dict[str, Any
-    asset = AssetField(required=True, expected_type=Asset, form_with_incomplete_data=True)
-    balance = fields.Nested(BalanceSchema, required=True)
-    location_label = fields.String(required=False)
-    notes = fields.String(required=False)
-    event_subtype = SerializableEnumField(
-        enum_class=HistoryEventSubType,
-        required=False,
-        load_default=HistoryEventSubType.NONE,
-        allow_none=True,
-    )
-    counterparty = fields.String(load_default=None)
-    product = SerializableEnumField(enum_class=EvmProduct, load_default=None)
-    address = EvmAddressField(load_default=None)
+    include_identifier: bool = False
+    entry_type = SerializableEnumField(enum_class=HistoryBaseEntryType, required=True)
+
+    class BaseSchema(Schema):
+        # Timestamp coming in from the API is in seconds, in contrast to what we save in the struct
+        timestamp = TimestampField(ts_multiplier=1000, required=True)
+        balance = fields.Nested(BalanceSchema, required=True)
+        identifier = fields.Integer(required=True)
+
+    class BaseEventSchema(BaseSchema):
+        event_type = SerializableEnumField(enum_class=HistoryEventType, required=True)
+        event_subtype = SerializableEnumField(
+            enum_class=HistoryEventSubType,
+            required=False,
+            load_default=HistoryEventSubType.NONE,
+            allow_none=True,
+        )
+        asset = AssetField(required=True, expected_type=Asset, form_with_incomplete_data=True)
+        notes = fields.String(load_default=None)
+        sequence_index = fields.Integer(required=True)
+        location_label = fields.String(load_default=None)
+
+    class CreateBaseHistoryEventSchema(BaseEventSchema):
+        event_identifier = fields.String(required=True)
+        location = LocationField(required=True)
+
+        @post_load
+        def make_history_base_entry(
+                self,
+                data: dict[str, Any],
+                **_kwargs: Any,
+        ) -> dict[str, Any]:
+            if data.get('event_subtype') is None:
+                data['event_subtype'] = HistoryEventSubType.NONE
+            return {'event': HistoryEvent(**data)}
+
+    class CreateEvmEventSchema(BaseEventSchema):
+        """Schema used when adding a new event in the EVM transactions view"""
+        tx_hash = EVMTransactionHashField(required=True)
+        counterparty = fields.String(load_default=None)
+        product = SerializableEnumField(enum_class=EvmProduct, load_default=None)
+        address = EvmAddressField(load_default=None)
+        extra_data = fields.Dict(load_default=None)
+        location = LocationField(required=True, limit_to=EVM_LOCATIONS)
+
+        @post_load
+        def make_history_base_entry(
+                self,
+                data: dict[str, Any],
+                **_kwargs: Any,
+        ) -> dict[str, Any]:
+            if data.get('event_subtype') is None:
```

Comment:
Perhaps time to change this. This used to be from before and is no longer needed since we are making changes here. Let's be explicit. Tell the frontend to send the special value `NONE` and that `null` is no longer a valid value. This way we can get rid of this special rule.